src/dice_builder_builder.rs

```diff
@@ -0,0 +1,24 @@
+use crate::dice_builder;
+pub struct DiceBuilderBuilder {
+    pub dice_builder_sides: u32,
+    pub dice_builder_name: String,
+}
+
+impl DiceBuilderBuilder {
+    pub fn sides(&mut self, dice_sides: u32) -> &mut Self {
+        self.dice_builder_sides = dice_sides;
+        self
+    }
+
+    pub fn name(&mut self, dice_name: String) -> &mut Self {
+        self.dice_builder_name = dice_name;
+        self
+    }
```

Comment:
You should take self here, so
- it's harder to call
- you need to own it fully